--- a/braille_puzzles.c
+++ b/braille_puzzles.c
@@ -267,13 +267,17 @@
 
     if (sIsRegisteelPuzzle == TRUE)
     {
-        gTasks[taskId].data[8] = (u32)UseRegisteelHm_Callback >> 16;
-        gTasks[taskId].data[9] = (u32)UseRegisteelHm_Callback;
+        gTasks[taskId].data[8] = (uintptr_t)UseRegisteelHm_Callback;
+        gTasks[taskId].data[9] = (uintptr_t)UseRegisteelHm_Callback >> 16;
+        gTasks[taskId].data[10] = (uintptr_t)UseRegisteelHm_Callback >> 32;
+        gTasks[taskId].data[11] = (uintptr_t)UseRegisteelHm_Callback >> 48;
     }
     else
     {
-        gTasks[taskId].data[8] = (u32)UseRegirockHm_Callback >> 16;
-        gTasks[taskId].data[9] = (u32)UseRegirockHm_Callback;
+        gTasks[taskId].data[8] = (uintptr_t)UseRegirockHm_Callback;
+        gTasks[taskId].data[9] = (uintptr_t)UseRegirockHm_Callback >> 16;
+        gTasks[taskId].data[10] = (uintptr_t)UseRegirockHm_Callback >> 32;
+        gTasks[taskId].data[11] = (uintptr_t)UseRegirockHm_Callback >> 48;
     }
     return FALSE;
 }
