--- a/easy_chat.c
+++ b/easy_chat.c
@@ -1299,8 +1299,8 @@
     taskId = CreateTask(Task_InitEasyChatScreen, 0);
     gTasks[taskId].tType = type;
     gTasks[taskId].tPersonType = displayedPersonType;
-    SetWordTaskArg(taskId, TASKIDX_WORDS, (u32)words);
-    SetWordTaskArg(taskId, TASKIDX_EXIT_CALLBACK, (u32)exitCallback);
+    SetWordTaskArg(taskId, TASKIDX_WORDS, (uintptr_t)words);
+    SetWordTaskArg(taskId, TASKIDX_EXIT_CALLBACK, (uintptr_t)exitCallback);
     SetMainCallback2(CB2_EasyChatScreen);
 }
 
