--- a/malloc.c
+++ b/malloc.c
@@ -4,7 +4,7 @@
 static void *sHeapStart;
 static u32 sHeapSize;
 
-ALIGNED(4) EWRAM_DATA u8 gHeap[HEAP_SIZE] = {0};
+ALIGNED(16) EWRAM_DATA u8 gHeap[HEAP_SIZE] = {0};
 
 #define MALLOC_SYSTEM_ID 0xA3A3
 
@@ -52,8 +52,8 @@
     u32 foundBlockSize;
 
     // Alignment
-    if (size & 3)
-        size = 4 * ((size / 4) + 1);
+    if (size & 7)
+        size = 8 * ((size / 8) + 1);
 
     for (;;)
     {
