--- a/scrcmd.c
+++ b/scrcmd.c
@@ -110,7 +110,7 @@
 
 bool8 ScrCmd_gotonative(struct ScriptContext *ctx)
 {
-    bool8 (*addr)(void) = (bool8 (*)(void))ScriptReadWord(ctx);
+    bool8 (*addr)(void) = (bool8 (*)(void))ScriptReadPtr(ctx);
 
     SetupNativeScript(ctx, addr);
     return TRUE;
@@ -134,7 +134,7 @@
 
 bool8 ScrCmd_callnative(struct ScriptContext *ctx)
 {
-    NativeFunc func = (NativeFunc)ScriptReadWord(ctx);
+    NativeFunc func = (NativeFunc)ScriptReadPtr(ctx);
 
     func();
     return FALSE;
@@ -148,7 +148,7 @@
 
 bool8 ScrCmd_goto(struct ScriptContext *ctx)
 {
-    const u8 *ptr = (const u8 *)ScriptReadWord(ctx);
+    const u8 *ptr = ScriptReadPtr(ctx);
 
     ScriptJump(ctx, ptr);
     return FALSE;
@@ -162,7 +162,7 @@
 
 bool8 ScrCmd_call(struct ScriptContext *ctx)
 {
-    const u8 *ptr = (const u8 *)ScriptReadWord(ctx);
+    const u8 *ptr = ScriptReadPtr(ctx);
 
     ScriptCall(ctx, ptr);
     return FALSE;
@@ -171,7 +171,7 @@
 bool8 ScrCmd_goto_if(struct ScriptContext *ctx)
 {
     enum ComparisonOperators condition = ScriptReadByte(ctx);
-    const u8 *ptr = (const u8 *)ScriptReadWord(ctx);
+    const u8 *ptr = ScriptReadPtr(ctx);
 
     if (sScriptConditionTable[condition][ctx->comparisonResult] == 1)
         ScriptJump(ctx, ptr);
@@ -181,7 +181,7 @@
 bool8 ScrCmd_call_if(struct ScriptContext *ctx)
 {
     enum ComparisonOperators condition = ScriptReadByte(ctx);
-    const u8 *ptr = (const u8 *)ScriptReadWord(ctx);
+    const u8 *ptr = ScriptReadPtr(ctx);
 
     if (sScriptConditionTable[condition][ctx->comparisonResult] == 1)
         ScriptCall(ctx, ptr);
@@ -306,7 +306,7 @@
 {
     u8 index = ScriptReadByte(ctx);
 
-    ctx->data[index] = ScriptReadWord(ctx);
+    ctx->data[index] = (uintptr_t)ScriptReadPtr(ctx);
     return FALSE;
 }
 
@@ -314,7 +314,7 @@
 {
     u8 index = ScriptReadByte(ctx);
 
-    ctx->data[index] = *(const u8 *)ScriptReadWord(ctx);
+    ctx->data[index] = *(const u8 *)ScriptReadPtr(ctx);
     return FALSE;
 }
 
@@ -322,7 +322,7 @@
 {
     u8 value = ScriptReadByte(ctx);
 
-    *(u8 *)ScriptReadWord(ctx) = value;
+    *(u8 *)ScriptReadPtr(ctx) = value;
     return FALSE;
 }
 
@@ -338,7 +338,7 @@
 {
     u8 index = ScriptReadByte(ctx);
 
-    *(u8 *)ScriptReadWord(ctx) = ctx->data[index];
+    *(u8 *)ScriptReadPtr(ctx) = ctx->data[index];
     return FALSE;
 }
 
@@ -353,8 +353,8 @@
 
 bool8 ScrCmd_copybyte(struct ScriptContext *ctx)
 {
-    u8 *ptr = (u8 *)ScriptReadWord(ctx);
-    *ptr = *(const u8 *)ScriptReadWord(ctx);
+    u8 *ptr = (u8 *)ScriptReadPtr(ctx);
+    *ptr = *(const u8 *)ScriptReadPtr(ctx);
     return FALSE;
 }
 
@@ -409,7 +409,7 @@
 bool8 ScrCmd_compare_local_to_ptr(struct ScriptContext *ctx)
 {
     const u8 value1 = ctx->data[ScriptReadByte(ctx)];
-    const u8 value2 = *(const u8 *)ScriptReadWord(ctx);
+    const u8 value2 = *(const u8 *)ScriptReadPtr(ctx);
 
     ctx->comparisonResult = Compare(value1, value2);
     return FALSE;
@@ -417,7 +417,7 @@
 
 bool8 ScrCmd_compare_ptr_to_local(struct ScriptContext *ctx)
 {
-    const u8 value1 = *(const u8 *)ScriptReadWord(ctx);
+    const u8 value1 = *(const u8 *)ScriptReadPtr(ctx);
     const u8 value2 = ctx->data[ScriptReadByte(ctx)];
 
     ctx->comparisonResult = Compare(value1, value2);
@@ -426,7 +426,7 @@
 
 bool8 ScrCmd_compare_ptr_to_value(struct ScriptContext *ctx)
 {
-    const u8 value1 = *(const u8 *)ScriptReadWord(ctx);
+    const u8 value1 = *(const u8 *)ScriptReadPtr(ctx);
     const u8 value2 = ScriptReadByte(ctx);
 
     ctx->comparisonResult = Compare(value1, value2);
@@ -435,8 +435,8 @@
 
 bool8 ScrCmd_compare_ptr_to_ptr(struct ScriptContext *ctx)
 {
-    const u8 value1 = *(const u8 *)ScriptReadWord(ctx);
-    const u8 value2 = *(const u8 *)ScriptReadWord(ctx);
+    const u8 value1 = *(const u8 *)ScriptReadPtr(ctx);
+    const u8 value2 = *(const u8 *)ScriptReadPtr(ctx);
 
     ctx->comparisonResult = Compare(value1, value2);
     return FALSE;
@@ -993,7 +993,7 @@
 bool8 ScrCmd_applymovement(struct ScriptContext *ctx)
 {
     u16 localId = VarGet(ScriptReadHalfword(ctx));
-    const void *movementScript = (const void *)ScriptReadWord(ctx);
+    const void *movementScript = (const void *)ScriptReadPtr(ctx);
 
     ScriptMovement_StartObjectMovementScript(localId, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, movementScript);
     sMovingNpcId = localId;
@@ -1003,7 +1003,7 @@
 bool8 ScrCmd_applymovementat(struct ScriptContext *ctx)
 {
     u16 localId = VarGet(ScriptReadHalfword(ctx));
-    const void *movementScript = (const void *)ScriptReadWord(ctx);
+    const void *movementScript = (const void *)ScriptReadPtr(ctx);
     u8 mapGroup = ScriptReadByte(ctx);
     u8 mapNum = ScriptReadByte(ctx);
 
@@ -1265,7 +1265,7 @@
 
 bool8 ScrCmd_message(struct ScriptContext *ctx)
 {
-    const u8 *msg = (const u8 *)ScriptReadWord(ctx);
+    const u8 *msg = ScriptReadPtr(ctx);
 
     if (msg == NULL)
         msg = (const u8 *)ctx->data[0];
@@ -1275,7 +1275,7 @@
 
 bool8 ScrCmd_pokenavcall(struct ScriptContext *ctx)
 {
-    const u8 *msg = (const u8 *)ScriptReadWord(ctx);
+    const u8 *msg = ScriptReadPtr(ctx);
 
     if (msg == NULL)
         msg = (const u8 *)ctx->data[0];
@@ -1285,7 +1285,7 @@
 
 bool8 ScrCmd_messageautoscroll(struct ScriptContext *ctx)
 {
-    const u8 *msg = (const u8 *)ScriptReadWord(ctx);
+    const u8 *msg = ScriptReadPtr(ctx);
 
     if (msg == NULL)
         msg = (const u8 *)ctx->data[0];
@@ -1298,7 +1298,7 @@
 // Prints all at once. Skips waiting for player input. Only used by link contests
 bool8 ScrCmd_messageinstant(struct ScriptContext *ctx)
 {
-    const u8 *msg = (const u8 *)ScriptReadWord(ctx);
+    const u8 *msg = ScriptReadPtr(ctx);
 
     if (msg == NULL)
         msg = (const u8 *)ctx->data[0];
@@ -1481,7 +1481,7 @@
 
 bool8 ScrCmd_braillemessage(struct ScriptContext *ctx)
 {
-    u8 *ptr = (u8 *)ScriptReadWord(ctx);
+    u8 *ptr = (u8 *)ScriptReadPtr(ctx);
     struct WindowTemplate winTemplate;
     s32 i;
     u8 width, height;
@@ -1645,7 +1645,7 @@
 bool8 ScrCmd_bufferstring(struct ScriptContext *ctx)
 {
     u8 stringVarIndex = ScriptReadByte(ctx);
-    const u8 *text = (u8 *)ScriptReadWord(ctx);
+    const u8 *text = (u8 *)ScriptReadPtr(ctx);
 
     StringCopy(sScriptStringVars[stringVarIndex], text);
     return FALSE;
@@ -1886,7 +1886,7 @@
 
 bool8 ScrCmd_pokemart(struct ScriptContext *ctx)
 {
-    const void *ptr = (void *)ScriptReadWord(ctx);
+    const void *ptr = (void *)ScriptReadPtr(ctx);
 
     CreatePokemartMenu(ptr);
     ScriptContext_Stop();
@@ -1895,7 +1895,7 @@
 
 bool8 ScrCmd_pokemartdecoration(struct ScriptContext *ctx)
 {
-    const void *ptr = (void *)ScriptReadWord(ctx);
+    const void *ptr = (void *)ScriptReadPtr(ctx);
 
     CreateDecorationShop1Menu(ptr);
     ScriptContext_Stop();
@@ -1905,7 +1905,7 @@
 // Changes clerk dialogue slightly from above. See MART_TYPE_DECOR2
 bool8 ScrCmd_pokemartdecoration2(struct ScriptContext *ctx)
 {
-    const void *ptr = (void *)ScriptReadWord(ctx);
+    const void *ptr = (void *)ScriptReadPtr(ctx);
 
     CreateDecorationShop2Menu(ptr);
     ScriptContext_Stop();
